### ISSUE TITLE
Remove body from HTTP error

### ISSUE THREAD

**Issue by rylev** [Azure/azure-sdk-for-rust#289]
`azure_core::UnexpectedStatusCode` error contains a field [`body`](https://github.com/Azure/azure-sdk-for-rust/blob/16bcf0ab1bb6e380d966a69d314de1e99ede553a/sdk/core/src/errors.rs#L92) which [we don't want to include](https://github.com/Azure/azure-sdk-for-rust/pull/286#discussion_r649414552) since this may expose secrets. This will probably make debugging harder though since on errors, the user will only get the status code and no indication from the body of what went wrong.

**Comment by MindFlavor** [comment]
Can we at least make it optional? We already have evidence (see https://github.com/Azure/azure-sdk-for-rust/issues/212) that the error body can be useful.

Maybe we could gate it behind a compilation flag? By default we would not show the error body but - for example - a "debug" version could.

What do you think?

**Comment by ctaggart** [comment]
I added it 27 days ago to fix #212 which requested a way to see it. I understand the argument for hiding it by default. Will have to think about this more.

**Comment by rylev** [comment]
@heaths how do other language SDKs handle this? The body can be extremely useful for see more context related to the error, but it might contain sensitive information the user may not wish to log.

**Comment by heaths** [comment]
Different languages do "attach" the body, but we don't log the body by default for any response. Errors also go through the same obfuscation process, at least for headers. @JoshLove-msft off hand, we don't do any extra obfuscation for error bodies though, right? We just don't log it by default?

**Comment by JoshLove-msft** [comment]
Correct, we just don't log it by default. If content logging is enabled, it is logged at Informational level and not obfuscated.
